### PR TITLE
Add CompiledBindings page

### PR DESCRIPTION
**Pull request by timunie**

### CODE REVIEW

File: docs/data-binding/compiledbindings.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Compiled Bindings
+
+Bindings defined in the XAML are using reflection in order to find and access the requested property in your `ViewModel`. In Avalonia you can also use compiled bindings, which has some benefits:
+  * If you use compiled bindings and the property you bind to is not found, the application will crash. Hence you get a much better debugging experience.
```
Comment (Takoooooo) [inline docs/data-binding/compiledbindings.md:4]: Not crash but you will get a compile-time error I think?

File: docs/data-binding/compiledbindings.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  * Reflection is known to be slow ([see this article on codeproject.com](https://www.codeproject.com/Articles/1161127/Why-is-reflection-slow)). Using compiled bindings can therefore improve the performance of your application.
```
Comment (Takoooooo) [inline docs/data-binding/compiledbindings.md:5]: Not sure, can we reference codeproject in our docs? @AvaloniaUI/core

File: docs/data-binding/compiledbindings.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Compiled Bindings
+
+Bindings defined in the XAML are using reflection in order to find and access the requested property in your `ViewModel`. In Avalonia you can also use compiled bindings, which has some benefits:
+  * If you use compiled bindings and the property you bind to is not found, the application will crash. Hence you get a much better debugging experience.
+  * Reflection is known to be slow ([see this article on codeproject.com](https://www.codeproject.com/Articles/1161127/Why-is-reflection-slow)). Using compiled bindings can therefore improve the performance of your application. 
+ 
+## Enable and disable compiled bindings
+
+Compiled bindings are not enabled by default. To enable compiled bindings, you will need to define the `DataType` of your `ViewModel` first. In [`DataTemplates`](https://docs.avaloniaui.net/misc/wpf/datatemplates) there is a property `DataType`, for all other elements you can set it via `x:DataType`. Most likely you will set `x:DataType` in your root node, for example in a `Window` or an `UserControl`.
```
Comment (Takoooooo) [inline docs/data-binding/compiledbindings.md:9]: I suggest not mentioning ViewModel here but the type of the object that will be used in pair with the view\style

File: docs/data-binding/compiledbindings.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                Command="{Binding SendEmailCommand}" />
+    </StackPanel>
+</UserControl>
+```
+
+## Known limitations
+
+Compiled bindings have some known limitations: 
+* Compiled bindings cannot be used to bind to methods ([see issue #4237](https://github.com/AvaloniaUI/Avalonia/issues/4237)).
+* Compiled bindings cannot be used to bind to named elements
```
Comment (Takoooooo) [inline docs/data-binding/compiledbindings.md:67]: I think there are more limitations,i will send you tomorrow the complete list.

File: docs/data-binding/compiledbindings.md (hunk 2 above)
Comment (kekekeks) [reply docs/data-binding/compiledbindings.md:5]: I don't think they've ever broken an article link, so it should be OK.

**Review by Arhell — APPROVED** [review]

File: docs/data-binding/compiledbindings.md (hunk 2 above)
Comment (timunie) [reply docs/data-binding/compiledbindings.md:5]: I just wanted to show an article, where the user can validate the information. I personally don't like to read such a statement, if I can't validate it. But if you want, I can also link [this issue](https://github.com/AvaloniaUI/Avalonia/issues/2359) instead or additionally.

**Comment by maxkatz6** [comment]
Worth to mention:
1. You can enabled compiled bindings for a file, but use ```{ReflectionBinding}``` where compiled doesn't work.
2. You can specify DataType inside of the markup extension ```"{Binding Path, DataType={x:Type MyClass}}"``` (will be available in next release, but you can add it now)

File: docs/data-binding/compiledbindings.md (hunk 4 above)
Comment (Takoooooo) [reply docs/data-binding/compiledbindings.md:67]: Sorry, I've forgotten about that. There is one additional limitation besides the ones you mentioned.
Bindings with RelativeSource TemplatedParent in the Style
```markup
{Binding Width, RelativeSource={RelativeSource TemplatedParent}}
```

**Comment by timunie** [comment]
@Takoooooo , @maxkatz6 

I have updated the docs now with your feedback. As I am not a native English speaker the text may sound rough to you. If you have more feedback, let me know 

Have a nice weekend
Tim

**Pull request merged**